src/parser-includes.ts

```diff
@@ -16,18 +16,26 @@ type ParserIncludesInitOptions = {
     fetchIncludes: boolean;
     excludedGlobs: string[];
     variables: {[key: string]: string};
+    isGitlabCiFile: boolean;
 };
 
-export class ParserIncludes {
+const MAXIMUM_INCLUDE = 150; // https://docs.gitlab.com/ee/administration/settings/continuous_integration.html#maximum-includes
 
-    static async init (gitlabData: any, depth: number, opts: ParserIncludesInitOptions): Promise<any[]> {
+export class ParserIncludes {
+    static count: number = 0;
+
+    static async init (gitlabData: any, opts: ParserIncludesInitOptions): Promise<any[]> {
+        if (opts.isGitlabCiFile) {
+            ParserIncludes.count++;
+            assert(
+                ParserIncludes.count <= MAXIMUM_INCLUDE + 1, // 1st call from parser.ts is not counted
+                chalk`This GitLab CI configuration is invalid: Maximum of {blueBright ${MAXIMUM_INCLUDE}} nested includes are allowed!.`
+            );
+        }
         let includeDatas: any[] = [];
         const promises = [];
         const {stateDir, cwd, fetchIncludes, gitData, excludedGlobs} = opts;
 
-        assert(depth < 100, chalk`circular dependency detected in \`include\``);
```

Comment:
removed the `depth` check, since the circular dependency will be caught by the `MAXIMUM_INCLUDE`